schemas/tests.yaml

```diff
@@ -0,0 +1,158 @@
+---
+
+#
+# JSON Schema definition for tmt `Tests`
+# https://tmt.readthedocs.io/en/stable/spec/tests.html
+#
+
+definitions:
+
+  beakerlib_library:
+    type: array
+    minItems: 1
+
+    items:
+      oneOf:
+        # library(repo/lib)
+        - type: string
+          pattern: "^library([a-zA-Z0-9]+/[a-zA-Z0-9]+)"
+
+        # url: https://github.com/beakerlib/openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+
+            url:
+              type: string
+              pattern: "^https?://"
+
+          required:
+            - name
+            - url
+
+        # path: /tmp/library/openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+
+            path:
+              type: string
+              pattern: "^/.+"
+
+          required:
+            - name
+            - path
+
+        # url: https://github.com/redhat-qe-security/certgen
+        # ref: devel
+        # nick: openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+            nick:
+              type: string
+            ref:
+              type: string
+            url:
+              type: string
+              pattern: "^https?://"
+
+          required:
+            - name
+            - nick
+            - ref
+            - url
+
+  string_or_array:
+    oneOf:
+      - type: string
+
+      - type: array
+        minItems: 1
```

Comment:
* I've seen `recommend`  and `contact` to be set to an empty list, i.e. `contact: []` is out there and not against the specification, therefore `minItems` seems way too restrictive to me.
* I'd add an item type, i.e.
```yaml
- type: array
   items:
     type: string
```

Reply:
it is up for discussion, I would rather have it restricted myself

Reply:
If I'm not mistaken, one might even want to use `require: []` or `recommend: []` on purpose, to undo whatever the (virtual) test inherited from its parent, is that right?

Reply:
I see, right, that is a valid use case I guess

Reply:
removed minItems